### PR TITLE
Upload recovery tool

### PR DESCRIPTION
**Pull request by cfarmer-fearless**

### CODE REVIEW

File: reupload-tool/src/main/kotlin/Main.kt [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        Reupload(
+            it["src"]!!,
+            it["dest"]!!,
+            it["srcaccountid"]!!
+        )
+    }
+}
+
+fun getFileFromResources(filename: String): File {
+    return File(object {}::class.java.classLoader.getResource(filename)?.file!!)
```
Comment (USZ7) [inline reupload-tool/src/main/kotlin/Main.kt:101]: Where does the class loader get the input.csv file from?

Comment (cfarmer-fearless) [reply reupload-tool/src/main/kotlin/Main.kt:101]: From the `resource` directory.

File: reupload-tool/src/main/kotlin/Main.kt [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+import com.github.doyaaaaaken.kotlincsv.dsl.csvReader
+import org.example.model.Reupload
+import org.example.util.EnvConfig
+import tus.UploadClient
+import java.io.File
+import kotlin.collections.HashMap
+
+fun main() {
+    // First, read in input.csv.
+    val inputCsv = getFileFromResources("input.csv")
```
Comment (whytheplatypus) [inline reupload-tool/src/main/kotlin/Main.kt:17]: get this from input, probably a flag with this as a default.

File: reupload-tool/src/main/kotlin/Main.kt [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    // Initialize auth client
+    val dexUrl = when(EnvConfig.DEX_ENV) {
+        "dev" -> "https://apidev.cdc.gov"
+        "tst" -> "https://apitst.cdc.gov"
+        "stg" -> "https://apistg.cdc.gov"
+        "prd" -> "https://api.cdc.gov"
+        else -> "https://apidev.cdc.gov"
+    }
+    val authClient = AuthClient(dexUrl)
+    val authToken = authClient.getToken(EnvConfig.SAMS_USERNAME, EnvConfig.SAMS_PASSWORD)
```
Comment (whytheplatypus) [inline reupload-tool/src/main/kotlin/Main.kt:30]: provide a way to skip auth

File: reupload-tool/src/main/kotlin/Main.kt [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+            println("downloading ${srcBlobClient.blobName} of size ${srcBlobClient.properties.blobSize}")
+            srcBlobClient.downloadToFile("downloads/$srcFilename")
+
+            // Overwrite filename in metadata with new filename
+            val updatedMetadata = updateFilename(reupload.dest, srcBlobClient.properties.metadata)
+
+            // Upload file.
+            val fileToUpload = File("downloads/$srcFilename")
+            println("uploading $fileToUpload")
+            uploadClient.uploadFile(fileToUpload, updatedMetadata)
```
Comment (whytheplatypus) [inline reupload-tool/src/main/kotlin/Main.kt:78]: suggestion: extract this and have it return a `Result`

**Review by whytheplatypus — CHANGES_REQUESTED** [review]

File: reupload-tool/src/main/kotlin/Main.kt (hunk 2 above)
Comment (cfarmer-fearless) [reply reupload-tool/src/main/kotlin/Main.kt]: Done.  Used Java system args for compatibility.

File: reupload-tool/src/main/kotlin/Main.kt (hunk 3 above)
Comment (cfarmer-fearless) [reply reupload-tool/src/main/kotlin/Main.kt:30]: Added a flag

File: reupload-tool/src/main/kotlin/Main.kt (hunk 4 above)
Comment (cfarmer-fearless) [reply reupload-tool/src/main/kotlin/Main.kt]: Done

**Comment by cfarmer-fearless** [comment]
Closing in favor of https://github.com/CDCgov/data-exchange-upload/pull/464

**Pull request closed without merging**